File: app/src/main/java/com/example/delivery/room/DaoFavorites.java
package com.example.delivery.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Entity;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.delivery.data.MyEntity;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface DaoFavorites {
    @Insert
    long saveFavorites (MyEntity myEntity);

    @Insert
    void saveList (MyEntity... myEntity);

    @Query("SELECT * FROM myentity")
    List<MyEntity> getFavorites();

    @Delete
    Single<Integer> deleteFavorite(MyEntity myEntity);

    @Query("SELECT * FROM myentity WHERE name=:name")
    MyEntity cheakPosition(String name);
}

File: app/src/main/java/com/example/deliverykotlin/room/DaoFavorites.kt
package com.example.deliverykotlin.room

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.example.deliverykotlin.data.Favorites
import com.example.deliverykotlin.data.MyEntity

@Dao
interface DaoFavorites {
    @Insert
    fun saveFavoritePosition(favorite: Favorites)

    @Query("SELECT * FROM myentity INNER JOIN favorites ON myentity.entityId=favorites.entity")
    fun getFavorites(): List<MyEntity>

    @Query("SELECT * FROM favorites WHERE favorites.entity=:id")
    fun getOneFavorite(id :Int):Favorites

    @Query("DELETE FROM favorites WHERE favorites.entity=:id")
    fun deleteOne(id: Int)

}